userdocs/diagnostics/expression-too-complex.md

```diff
@@ -0,0 +1,16 @@
+# Type Inference in Complex Expressions
+For some complex expressions, the Swift compiler is unable to infer types in reasonable time. This can occur for a number of reasons, but some common ones include:
```

Comment:
Nit: We're really defining what it means for an expression to be "too complex" here, so perhaps best to avoid using the term in the first sentence of the definition. (Though, strictly, we're defining "_too_ complex" rather than "complex"; but the idea is the same.)

```suggestion
For some expressions, the Swift compiler is unable to infer types in reasonable time. This can occur for a number of reasons, but some common ones include:
```